Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
+{
+    GameObject gameObject;
+    EventController.Action action; 
+    bool isDown;
+    public delegate void performAction (EventController.Action action);  
+
+    public UIButton (GameObject gameObject, EventController.Action action)
+    {
+        this.gameObject = gameObject;
+        this.action = action; 
+    }
+
+    public void ButtonDown()
+    {
+        isDown = true;
+        AddButtonHeldEvent(action);     
+    }
+
+    public void ButtonUp()
```

Comment:
I'll stop making the same comment about `void CallActions()`, but would recommend applying that pattern across the board